packages/dynamite/dynamite/lib/src/builder/generate_ofs.dart

```diff
@@ -112,8 +137,9 @@ Iterable<Spec> generateSomeOf(
 
   yield Extension(
     (final b) => b
-      ..name = '${identifier}Extension'
+      ..name = '${identifier}Extension'.nonPrivate
       ..on = refer(identifier)
+      ..docs.add('// ignore: library_private_types_in_public_api')
```

Comment:
The whole extension could be private so we don't need this, right?

Reply:
Then we also wouldn't need to hash the name as it won't appear in the docs anyway :thinking:

Reply:
This would also hide the `toJson` and `validate` methods` which might be interesting to devs.

Reply:
But we could expose those through the proxy extension? I like this way more as it will solve this problem the right way imo

Reply:
But we could expose those through the proxy extension? 
Nope as you would have multiple extension declaring the same method.

Which one should be used?
```dart
print('value'.name);

extension Extension1 on String{
  String get name => '1';
}

extension Extension2 on String{
  String get name => '2';
}
```
I know that one could explicitly specify this like `Extension1('value').name`. But that's tedious.

Reply:
Ah I see :/
I would still prefer to make the whole extension private. We expose the serializers so if someone really needs toJson they can do it manually.

Reply:
And we can still make it public later, it wouldn't be a breaking change. Keeping it public now and making it private later will be a breaking change.

Reply:
The seriailizers with all registered builder factories needed are still annotated with `visibleForTesting`.

Reply:
But that we can also change later without it being a breaking change if necessary.

Reply:
But I realized in this case we can also argue against the proxy extension and just make it all private. If serializing/deserializing is necessary then we need to expose the jsonSerializers.

Reply:
I think the current implementation is fine.
We can still add doc comments to the ugly extension later.

Reply:
But do we really want to have the hashed extensions appear in the docs? Is there maybe an annotation that removes it from the docs but makes it still accessible (not like `@internal`)?